tests/large_transactions_test.go

```diff
@@ -0,0 +1,184 @@
+
+func TestLargeTransactions_LargeTransactionLoadTest(t *testing.T) {
+	hardForks := map[string]opera.Upgrades{
+		"Sonic":   opera.GetSonicUpgrades(),
+		"Allegro": opera.GetAllegroUpgrades(),
+	}
+
+	modes := map[string]bool{
+		"DistributedProposer": false,
+		"SingleProposer":      true,
+	}
+
+	for name, upgrades := range hardForks {
+		for mode, singleProposer := range modes {
+			t.Run(fmt.Sprintf("%s/%s", name, mode), func(t *testing.T) {
+				effectiveUpgrades := upgrades
+				effectiveUpgrades.SingleProposerBlockFormation = singleProposer
+				testLargeTransactionLoadTest(t, &effectiveUpgrades)
+			})
+		}
+	}
+}
+
+func testLargeTransactionLoadTest(
+	t *testing.T,
+	upgrades *opera.Upgrades,
+) {
+	// The aim of this test is to flood the network with large transactions to
+	// trigger the production of messages exceeding the maximum limit of 10 MB.
+	// If this happens, events are not forwarded between nodes, leading to a
+	// network stall -- observable by the fact that the transactions are not
+	// processed and no receipts are produced. This test ensures that the
+	// network can handle such a load without stalling.
+	const (
+		numAccounts = 50
+		numRounds   = 10
+	)
+	require := require.New(t)
+	net := StartIntegrationTestNet(t, IntegrationTestNetOptions{
+		Upgrades: upgrades,
+		NumNodes: 3,
+	})
+
+	// Increase the gas limit to allow for larger transactions in blocks. These
+	// limits are beyond safe limits acceptable for production.
+	current := getNetworkRules(t, net)
+
+	modified := current.Copy()
+	modified.Economy.Gas.MaxEventGas = 1_000_000_000
+	modified.Economy.ShortGasPower.AllocPerSec = 20_000_000_000
+	modified.Economy.ShortGasPower.MaxAllocPeriod = 50_000_000_000
+	modified.Economy.LongGasPower = modified.Economy.ShortGasPower
+	modified.Emitter.Interval = 200_000_000 // low a bit down to provoke larger events
+	updateNetworkRules(t, net, modified)
+	require.NoError(net.AdvanceEpoch(1))
+
+	// Check that the modification was applied.
+	current = getNetworkRules(t, net)
+	require.Equal(modified, current)
+
+	// Create accounts and provide them with funds to run the load test.
+	accounts := make([]*Account, numAccounts)
+	addresses := make([]common.Address, len(accounts))
+	for i := range accounts {
+		accounts[i] = NewAccount()
+		addresses[i] = accounts[i].Address()
+	}
+	endowment := new(big.Int).Mul(big.NewInt(100), big.NewInt(1e18))
+	_, err := net.EndowAccounts(addresses, endowment)
+	require.NoError(err)
+
+	chainId := net.GetChainId()
+	signer := types.NewCancunSigner(chainId)
+
+	// Create a list of large transactions to flood the network.
+	transactions := []*types.Transaction{}
+	data := make([]byte, 125_000) // 125 KB of data, all zeros (cheapest)
+	for nonce := range uint64(numRounds) {
+		for i := range accounts {
+			tx := types.MustSignNewTx(
+				accounts[i].PrivateKey,
+				signer,
+				&types.AccessListTx{
+					ChainID:  chainId,
+					Gas:      125_000*10 + 21_000, // 125 KB of data + base gas
```

Comment:
maybe add a comment to clarify that `*10` is the price of zero bytes in data ? wdyt ?